Explorer/Assets/Scripts/ECS/StreamableLoading/DeferredLoading/DeferredLoadingSystem.cs

```diff
@@ -0,0 +1,91 @@
+using Arch.Core;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using ECS.Abstract;
+using ECS.Prioritization.Components;
+using ECS.StreamableLoading.AssetBundles;
+using ECS.StreamableLoading.AssetBundles.Manifest;
+using ECS.StreamableLoading.Common.Components;
+using ECS.StreamableLoading.Textures;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using Unity.Collections.LowLevel.Unsafe;
+using UnityEngine;
+
+namespace ECS.Prioritization.DeferredLoading
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [UpdateAfter(typeof(PrepareAssetBundleManifestParametersSystem))]
+    [UpdateBefore(typeof(LoadAssetBundleManifestSystem))]
+    [UpdateBefore(typeof(LoadTextureSystem))]
+    [UpdateBefore(typeof(LoadAssetBundleSystem))]
+    public partial class DeferredLoadingSystem<TAsset, TIntention> : BaseUnityLoopSystem where TIntention: struct, ILoadingIntention
+    {
+        private static readonly QueryDescription CREATE_LOADING_REQUEST = new QueryDescription()
+                                                                         .WithAll<TIntention>()
+                                                                         .WithNone<LoadingInProgress, StreamableLoadingResult<TAsset>>();
+        private readonly Query query;
+
+        private readonly IConcurrentBudgetProvider concurrentLoadingBudgetProvider;
+        private readonly unsafe List<IntentionData> loadingIntentions;
+
+        public DeferredLoadingSystem(World world, ConcurrentLoadingBudgetProvider concurrentLoadingBudgetProvider) : base(world)
+        {
+            this.concurrentLoadingBudgetProvider = concurrentLoadingBudgetProvider;
+            query = World.Query(in CREATE_LOADING_REQUEST);
+            loadingIntentions = new List<IntentionData>();
+        }
+
+        protected override unsafe void Update(float t)
+        {
+            loadingIntentions.Clear();
+
+            foreach (ref Chunk chunk in query.GetChunkIterator())
+            {
+                ref TIntention intentionFirstElement = ref chunk.GetFirst<TIntention>();
+                ref PartitionComponent partitionFirstElement = ref chunk.GetFirst<PartitionComponent>();
+
+                foreach (int entityIndex in chunk)
+                {
+                    ref PartitionComponent partition = ref Unsafe.Add(ref partitionFirstElement, entityIndex);
+                    ref TIntention intention = ref Unsafe.Add(ref intentionFirstElement, entityIndex);
```

Comment:
You call `Unsafe.Add` twice - this line is redundant. It can't work this way as you advance the pointer two times each iteration